packages/mui-joy/src/Select/Select.tsx

```diff
@@ -170,14 +168,8 @@ const SelectRoot = styled('div', {
         borderRadius: 'inherit',
         margin: 'calc(var(--variant-borderWidth, 0px) * -1)', // for outlined variant
       },
-      [`&.${selectClasses.focusVisible}`]: {
-        '--Select-indicatorColor': variantStyle?.color,
-        '&::before': {
-          boxShadow: `inset 0 0 0 var(--Select-focusedThickness) var(--Select-focusedHighlight)`,
-        },
-      },
-      [`&.${selectClasses.disabled}`]: {
-        '--Select-indicatorColor': 'inherit',
+      [`&:focus-within::before`]: {
+        boxShadow: `inset 0 0 0 var(--Select-focusedThickness) var(--Select-focusedHighlight)`,
```

Comment:
Out of curiosity - is there an advantage of using a hard box-shadow instead of an outline?

Reply:
I don't think so. `outline` should be used in this case (for focus).

Reply:
Wasn't there an issue with `outline` not following the container's border radius in Safari, I think? 🤔